### PR TITLE
[FS] Improve the Sidechain Dashboard

### PR DESCRIPTION
**Pull request by clintnetwork**
## Sidechain Dashboard Enhancement

screen2_censored

### Describing the PR

- Implement a voting system for the federation
- ~Add possiblity to set logging level of the nodes~
- ~Fix and update some assets~
- ~Little code reformat/cleaning~
- ~Update packages and assets~

### Some questions:

- ~Does I can make a .gitignore file in the Sidechain Dashboard project or I need to use the solution .gitignorefile ?~
- Does I need to add the Dashboard to the Solution file ?
- About the Crosschain Resync, does we implement this functions or not ?

**PS: Please keep opened for now**

Replace the #3407

### CODE REVIEW

**Comment by clintnetwork** [comment]
Question about .gitignore:

I want to ignore some minified files (*.min.css and *.min.js), do I use the .gitignore file in the root or in the Stratis.FederatedSidechains.AdminDashboard folder ?

Not sure that I can have a particular .gitignore and I can produce problems if I add elements in the root .gitignore file.

**Comment by clintnetwork** [comment]
Question about .gitignore:

I want to ignore some minified files (*.min.css and *.min.js), do I use the .gitignore file in the root or in the Stratis.FederatedSidechains.AdminDashboard folder ?

Not sure that I can have a particular .gitignore and I can produce problems if I add elements in the root .gitignore file.

**Comment by clintnetwork** [comment]
Does I need to add the Sidechain Dashboard to the FN Solution ?

image

**Comment by codingupastorm** [comment]
Hey @clintnetwork sorry I haven't got to this yet! I know you asked for a review.  Been meaning to review it for some time. I should get some time today, so will get back to you :)

**Comment by clintnetwork** [comment]
Hi @codingupastorm, I see that we are on the same time zone :p

I don't really asked for a review but i've some question, if I could have answers that could be great

- Does I can make a .gitignore file in the Sidechain Dashboard project or I need to use the solution .gitignorefile ?
- Does I need to add the Dashboard to the Solution file ?
- About the Crosschain Resync, does we implement this functions or not ?

**Comment by codingupastorm** [comment]
Looks like we are ;) 

1) No need for another gitignore - we have one for the whole repo that will work fine. If you need to edit it feel free.

2) Hmmm. Good question. Let me ask Iain.

3) I'm not sure what `Crosschain Resync` is. I might need to ask @quantumagi

**Comment by quantumagi** [comment]
There is a wallet resync (or removal of wallet transactions). Crosschain resync however is not implemented as far as I know. I also don't see any immediate need for a resync as I believe this can be done manually similar to one would to a block chain resync - i.e. by deleting the cross-chain store folder. @rowandh , perhaps you can also comment on this?

File: .gitignore [hunk 1]
```diff
@@ -254,3 +254,6 @@ ModelManifest.xml
 /src/Stratis.StratisD/Logs
 /src/Stratis.BitcoinD/Logs
 /src/Stratis.Benchmark/BenchmarkDotNet.Artifacts/results
+/src/Stratis.FederatedSidechains.AdminDashboard/wwwroot/css/*.min.css
```
Comment (clintnetwork) [inline .gitignore]: I migrate some rules in the whole repo .gitignore

**Comment by clintnetwork** [comment]
@quantumagi so I need to comment the crosschain resyncing ?

image

**Comment by clintnetwork** [comment]
Yes maybe

File: .gitignore (hunk 1 above)
Comment (codingupastorm) [reply .gitignore]: Nice.

**Comment by codingupastorm** [comment]
@clintnetwork @quantumagi I'll wait for some feedback from the testing team and if things are dire we can make a decision then on whether to include this.

My understanding it was put in there as a manual fix when the wallets "broke / got out of sync". We really should be able to avoid that altogether, but maybe someone who designed the system knows something we don't haha.

So... for now, leave it in or take it out, and we'll get back to you soon @clintnetwork .

**Comment by codingupastorm** [comment]
Also heard back from Iain, we don't need to add this to the solution, but so long as it's in the repo that's great. Hopefully the API doesn't change and nothing gets broken :)

**Comment by clintnetwork** [comment]
Yes for the changes on the API it's a possibility I hope to that never will change, for the crosschain resync I will comment the link in the view.

I think all is good, just waiting the PoA network to try the API and implement this feature.

Feel feel to think about new features that we can add to the sidechain dashboard, (Iain asked me)

**Comment by clintnetwork** [comment]
@codingupastorm @quantumagi @rowandh, you can merge the PR, you can squash it if you want.

@StratisIain is ok, I will make a new PR for the PoA voting part.

File: src/Stratis.Bitcoin.FullNode.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27130.2027
+# Visual Studio Version 16
```
Comment (clintnetwork) [inline src/Stratis.Bitcoin.FullNode.sln]: Please check if the version is good

File: src/Stratis.Bitcoin.FullNode.sln [hunk 3]
```diff
@@ -158,15 +158,17 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.Features.FederatedP
...
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.CirrusPegD", "Stratis.CirrusPegD\Stratis.CirrusPegD.csproj", "{3E6D4D9E-CAF5-4A23-9C96-07B1B1FE76B7}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.Features.FederatedPeg.IntegrationTests", "Stratis.Features.FederatedPeg.IntegrationTests\Stratis.Features.FederatedPeg.IntegrationTests.csproj", "{D71A7BBE-E64A-4F2E-B2A9-6D742FE94ED9}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Benchmarks", "Benchmarks", "{7FFED910-450B-4D6E-9085-6E5A6731D72A}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Stratis.Benchmark", "Stratis.Benchmark\Stratis.Benchmark.csproj", "{B8BC6552-6409-4BE1-B085-E8E6513EC251}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.Benchmark", "Stratis.Benchmark\Stratis.Benchmark.csproj", "{B8BC6552-6409-4BE1-B085-E8E6513EC251}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.FederatedSidechains.AdminDashboard", "Stratis.FederatedSidechains.AdminDashboard\Stratis.FederatedSidechains.AdminDashboard.csproj", "{516F4F2E-587C-4C55-91A3-C552A34C0CC8}"
```
Comment (clintnetwork) [inline src/Stratis.Bitcoin.FullNode.sln]: I have added the Sidechain Dashboard project in the `Apps` folder

image

File: src/Stratis.Bitcoin.FullNode.sln (hunk 2 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.FullNode.sln]: Why change this at all?

Comment (dangershony) [reply src/Stratis.Bitcoin.FullNode.sln]: Please revert this change.

File: src/Stratis.FederatedSidechains.AdminDashboard/Program.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public static void Main(string[] args)
 
         public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
-                .UseUrls("http://localhost:1989")   //TODO: temporary port
+                .UseUrls("http://localhost:37000")
```
Comment (dangershony) [inline src/Stratis.FederatedSidechains.AdminDashboard/Program.cs]: After this commit you could potentially create a feature that will start this on the fed node startup.

File: src/Stratis.Bitcoin.FullNode.sln (hunk 2 above)
Comment (clintnetwork) [reply src/Stratis.Bitcoin.FullNode.sln]: it was automatically, I use VS2019

File: src/Stratis.FederatedSidechains.AdminDashboard/Program.cs (hunk 4 above)
Comment (clintnetwork) [reply src/Stratis.FederatedSidechains.AdminDashboard/Program.cs]: Why not, do you know the location or something that can help me ?

**Comment by clintnetwork** [comment]
Anybody can resolve the conflict ? I don't know what is the good fix

And also merge the PR

**Comment by dangershony** [comment]
yes I will resovle this

**Comment by clintnetwork** [comment]
image

oops, please merge when possible, I removed my fork I hope all will be good...

**Pull request merged**